### PR TITLE
Textfield, RadioGroup, SearchField:   implement background color tokens in form fields and some code cleanups 

### PR DESCRIPTION
**Pull request by AlbertCarreras**
## Breaking change

Run

```
yarn codemod detectManualReplacement ~/path/to/your/code
--component=SearchField
--prop=onChange
```
or

```
yarn codemod detectManualReplacement ~/path/to/your/code
--component=SearchField
--prop=onFocus
```

`onChange: ({   value: string,   syntheticEvent: SyntheticEvent<HTMLInputElement>, }) => void
`
to 

`onChange: ({   value: string,   event: SyntheticEvent<HTMLInputElement | HTMLButtonElement>, }) => void
`

```
onFocus?: ({
  syntheticEvent: SyntheticKeyboardEvent<HTMLInputElement>,
  value: string,
}) => void
```

to 

```
onFocus?: ({
    event: SyntheticKeyboardEvent<HTMLInputElement>,
    value: string,
  }) => void
```

#### What changed?

Textfield:   implement background color tokens in form fields and some code cleanups 

### MINOR

NEW

| name                                | light   | dark    |
| ----------------------------------- | ------- | ------- |
| color-background-formfield-primary  | #fff    | #030303 |
| color-background-formfield-disabled | #efefef | #404040 |

#### Why

Tokenize FormField component background colors

### CODE REVIEW

File: packages/gestalt-design-tokens/tokens/color/component.json [hunk 1]
```diff
@@ -346,6 +346,18 @@
           }
         }
       },
+      "formfield": {
+        "primary": {
+          "value": "#fff",
```
Comment (rlingineni) [inline packages/gestalt-design-tokens/tokens/color/component.json:351]: How come this isn't referring to our actual token? And it's hex values?

File: packages/gestalt/src/SearchField.css [hunk 2]
```diff
@@ -47,22 +46,13 @@ html[dir="rtl"] .inputActive {
 
 .clear {
```
Comment (rlingineni) [inline packages/gestalt/src/SearchField.css:47]: This needs visual snapshot updates no?

**Review by rlingineni — CHANGES_REQUESTED** [review]

File: packages/gestalt-design-tokens/tokens/color/component.json (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: Our lower level tokens don't cover this color.

Screenshot by Dropbox Capture


I'm prioritizing tokenizing vs creating lower level tokens which  should be design-driven

Comment (rlingineni) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: Isnt' this the original token?

```
. whiteBg {
 background-color: var(--g-colorGray0);
}
```

File: packages/gestalt/src/SearchField.css (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/SearchField.css:47]: The visual snapshot doesnt cover the "with content" state just the empty state. The button only shows up when there's input

Comment (AlbertCarreras) [reply packages/gestalt/src/SearchField.css:47]: I asked designers if they were ok with using IconButton here and shared visual changes. @cintiapub and @hectoid were ok with it as with VR and more squared roundings the icon sizing wont be as much rellevant

File: packages/gestalt-design-tokens/tokens/color/component.json (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: Screenshot by Dropbox Capture

Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: Screenshot by Dropbox Capture

Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: @rlingineni Yes, it's that one.

Comment (rlingineni) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: oh what, these aren't tokens?

**Review by rlingineni — APPROVED** [review]

File: packages/gestalt-design-tokens/tokens/color/component.json (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: Nope.

Comment (AlbertCarreras) [reply packages/gestalt-design-tokens/tokens/color/component.json:351]: I called them "legacy colors" or color variables or old colors haha

**Review by eniomoura — APPROVED** [review]

**Pull request merged**